src/main/common/filter.h

```diff
@@ -74,6 +84,12 @@ typedef struct biquadFilterVec3_s {
     float y1[XYZ_AXIS_COUNT], y2[XYZ_AXIS_COUNT];
 } biquadFilterVec3_t;
```

Comment:
I'd much rather see something like:
```c
typedef struct biquadFilterState_s {
    float x1, x2, y1, y2;
} biquadFilterState_t;

typedef struct biquadFilter_s {
    biquadFilterGains_t gains;
    biquadFilterState_t state[1];
} biquadFilter_t;

typedef struct biquadFilterVec3_s {
    biquadFilterGains_t gains;
    biquadFilterState_t state[XYZ_AXIS_COUNT];
} biquadFilterVec3_t;
```

etc

Also, I don't really like the term `gains` in this context. I think `coefficients` or `coeffs` is better. That is:

```c
typedef struct biquadFilterCoeffs_s {
    float b0, b1, b2, a1, a2;
} biquadFilterCoeffs_t;
```

Reply:
And then pass both biquadFilterState_t* and biquadFilterCoeffs_t* to inner filter function - no explicit index and no dependency on number of variables